src/turn-command-manager.ts

```diff
@@ -0,0 +1,467 @@
+      return false;
+    }
+
+    const turnCommand = this.tryRemoveCommand((tc) => tc.command === BattleCommand.FIGHT && commandFilter(tc));
+
+    if (turnCommand) {
+      const { pokemon, cursor, move: queuedMove, targets } = turnCommand;
+      if (!pokemon.isActive(true) || !queuedMove) {
+        return false;
+      }
+
+      pokemon.turnData.order = this.orderIndex++;
+
+      const move =
+        pokemon.getMoveset().find((m) => m.moveId === queuedMove.moveId && m.ppUsed < m.getMovePp())
+        ?? new PokemonMove(queuedMove.moveId);
+
+      globalScene.appendToPhase(
+        new MovePhase(pokemon, targets ?? queuedMove.targets, move, false, cursor !== -1 && queuedMove.ignorePP),
+        PhaseId.MOVE_END,
+      );
+
+      return true;
+    }
+    return false;
+  }
+
+  /** Schedules all turn commands to be run at the start of the turn. */
+  public startTurn(): void {
+    // Apply speed-bypassing effects for all remaining Pokemon
+    /** @todo Find a way to apply this after non-FIGHT commands are processed */
+    this.applyBypassSpeedEffects();
+    // Shuffle and sort turn commands by speed, command type, priority, etc.
+    this.setTurnOrder(false);
+    // Add all commands that aren't using moves to the phase queue
+    this.shiftNonFightCommands();
+    // Add all move header effects to the phase queue
+    this.applyMoveHeaderAttrs();
+    // Add the first valid move command to the phase queue.
+    // This loop ensures that skipped and invalid commands do not
+    // freeze the turn sequence.
+    while (!this.empty() && !this.shiftNextCommand());
+  }
+
+  public empty(): boolean {
+    return !this.turnCommands.length;
+  }
+
+  /**
+   * Changes the given Pokemon's queued command to instead use the given move
+   * against the given target(s).
+   * @param pokemon the {@linkcode Pokemon} whose command is modified
+   * @param move the {@linkcode PokemonMove} replacing the current command's move
+   * @param targets the {@linkcode BattlerIndex | targets} for the replacement move
+   * @returns `true` if a turn command was modified
+   */
+  public tryReplaceMove(pokemon: Pokemon, move: PokemonMove, targets: BattlerIndex[]): boolean {
+    const turnCommand = this.findPokemonCommand(pokemon);
+    if (turnCommand?.command !== BattleCommand.FIGHT) {
+      return false;
+    }
+
+    const newMove: QueuedMove = {
+      moveId: move.moveId,
+      targets,
+    };
+
+    turnCommand.move = newMove;
+    turnCommand.targets = targets;
+
+    return true;
+  }
+
+  // #region Private Methods
+
+  /** Randomly shuffles the turn command queue. */
+  private shuffle(): void {
+    // This is seeded with the current turn to prevent an inconsistency where it
+    // was varying based on how long since you last reloaded
+    globalScene.executeWithSeedOffset(
+      () => {
+        this.turnCommands = randSeedShuffle(this.turnCommands);
+      },
+      globalScene.currentBattle.turn,
```

Comment:
```suggestion
      globalScene.currentBattle.turn * 1000 + this.turnCommands.length,
```
Trying to prevent RNG correlation bugs from shuffling using the same seed and offset multiple times in the same turn. (This assumes that `turnCommands.length` decreases when a Pokemon uses a move.)